### PR TITLE
[Inductor] [Doc] Add debugging document for inductor cpu backend

### LINKED ISSUE #2348: Segfault when using together with Pandas
When both PyTorch and Pandas are imported, calling `loss.backward()` results in a segfault. The issue was originally reported by @EthanRosenthal [here](https://github.com/maciejkula/spotlight/issues/42)

It looks like calling `PyThreadState_GET()` returns a null pointer here: https://github.com/python/cpython/blob/5fd33b5926eb8c9352bf5718369b4a8d72c4bb44/Python/errors.c#L28

Examining in GDB:
```
(gdb) down
#0  PyErr_Restore (type=0x0, value=<optimised out>, traceback=0x0) at Python/errors.c:42
42      Python/errors.c: No such file or directory.
(gdb) info locals
tstate = 0x0
oldtype = <optimised out>
oldvalue = <optimised out>
oldtraceback = <optimised out>
(gdb) tstate
Undefined command: "tstate".  Try "help".
(gdb) p tstate
```
I suspect it's both PyTorch and Pandas doing threading magic and somehow getting in each other's way. 

I haven't yet been able to provide a minimal example that triggers this behaviour and does not use my package; it's likely that I'm doing something odd that causes this.

More documentation here: https://github.com/maciejkula/spotlight/issues/42
Possibly related issues: https://github.com/pytorch/pytorch/issues/1318 and https://github.com/pytorch/pytorch/issues/1868.

Python version: 3.6
PyTorch version: v0.2.0
Pandas version: 0.20.1

### LINKED ISSUE #2348: 💡 [REQUEST] - Write a Tutorial for Using, Debugging, Performance Profiling torch.compile with Inductor CPU backend
### 🚀 Descirbe the improvement or the new tutorial

PyTorch 2.0 introduced the flagship compilation API, `torch.compile`, which offers a significant speedup over eager mode execution through graph-level optimization powered by the default TorchInductor backend. While this new feature has generated excitement within the PyTorch community, there is a lack of comprehensive tutorials that delve into the intricacies of `torch.compile`. The existing tutorials primarily focus on basic usage while missing the essential aspects such as exploring the underlying generated code, debugging potential issues, and conducting performance profiling. Therefore, this proposal aims to address this gap by creating an in-depth tutorial specifically designed for the Inductor CPU backend.

### Existing tutorials on this topic

https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html

### Additional context

We aim to complete the document as part of PyTorch Docathon 2023. cc @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @ZailiWang @ZhaoqiongZ @leslie-fang-intel @Xia-Weiwen @sekahler2 @CaoE @zhuhaozhe @Valentine233 @EikanWang

### PR DESCRIPTION
**Pull request by Valentine233**
Fixes [#2348](https://github.com/pytorch/tutorials/issues/2348)

Add document for inductor debugging and performance profiling with cpu backend.

cc @zhuhaozhe @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by Valentine233** [comment]
@jgong5 @EikanWang Hi, please review for the draft document of inductor debugging. The part of profiling would be added soon. cc @zhuhaozhe

**Comment by msaroufim** [comment]
When you're done make sure to also link your rst here https://github.com/pytorch/pytorch/blob/main/docs/source/index.rst

**Comment by jgong5** [comment]
Shall we submit the doc to the tutorial repo instead? I saw a tutorial about `torch.compile` there: https://github.com/pytorch/tutorials/blob/main/intermediate_source/torch_compile_tutorial.py

**Comment by Valentine233** [comment]
> Shall we submit the doc to the tutorial repo instead? I saw a tutorial about `torch.compile` there: https://github.com/pytorch/tutorials/blob/main/intermediate_source/torch_compile_tutorial.py

Thanks. I will make a new PR to put the doc in the same directory as ``torch_compile_tutorial.py``.

**Comment by Valentine233** [comment]
Move to https://github.com/pytorch/tutorials/pull/2430.
